### DISCUSSION TITLE
Plugins Add Calendar requires time property dateModified, why?

### DISCUSSION THREAD

**Discussion by wewhite** [wet-boew/wet-boew#9444] [Q&A]
Why is the time property dateModified required for the Add Calendar Plugin?
`			
// Set date stamp with the date modified`
`event_details.dtStamp = dtToISOString( $( "time[property='dateModified']" ) );`

if this is missing, a RangeError: Invalid Date occurs in toISOString.

The event_details.dtStamp is then used as:
`// Set Unique Identifier (UID) and Date Stamp (DSTAMP)`
`event_details.uid = window.location.href.replace( /\.|-|\/|:|[G-Zg-z]/g, "" ).toUpperCase().substr( -10 ) + "-" + event_details.sDate + "-" + event_details.dtStamp;`

as well as:
`// Set ICS file for Outlook, Apple and other calendars`
`icsFile = "BEGIN:VCALENDAR\nVERSION:2.0\nPRODID:-//WET-BOEW//Add to Calendar v4.0//\nBEGIN:VEVENT\nDTSTAMP:" + event_details.dtStamp +...`

**Comment by wewhite** [comment]
I modified the date stamp to be the following:
`// Set date stamp with the current date`
`event_details.dtStamp = new Date().toISOString();`

Makes more sense to me, and also do not need 'dateModified' on page.

Should I create an issue/pull request?

**Comment by ghost** [comment]
I recommend posting an issue, but no. If using `new Date()` then the date always be the current day.

Please review https://wet-boew.github.io/v4.0-ci/demos/add-cal/add-cal-en.html.

###### I am just a volunteer on this GitHub project.

**Comment by wewhite** [comment]
Found it in the documentation:
https://wet-boew.github.io/v4.0-ci/docs/ref/add-cal/add-cal-en.html
**_The propery "datemodified" must be present in the web page._**

I'll leave it as is, until I run into an actual issue.
My pages do not have datemodified, I'll have to add that.

Thank you.

**Comment by duboisp** [comment]
@wewhite You can submit a PR which will implement an alternative fallback for when the `schema:dateModified` is not provided via RDFa markup in the current page as long the alternative date meet the requirement for DTSTAMP.

**DTSTAMP**: It specifies the date and time of when the event calendar was created. Its value MUST be specified in the UTC time format.
ref. https://www.rfc-editor.org/rfc/rfc5545#section-3.8.7.2

Usually in the page, there is other date information we can rely on, like by using the meta data of the page such `<meta name="dcterms.modified" content="2022-11-17">` 

And for situation where no date modified is available, I do think the plugin should throw an error in the browser console. The date modify used by the add calendar event also help calendar software to know if the event has been updated since the last time it was added.

If you need help, feel free to join us at anytime during our weekly Tuesday office hours from 1pm to 4pm Eastern time. All the information and link are available here: https://github.com/wet-boew/wet-boew/wiki/WET-Office-hours,-Heures-de-service-de-la-BOEW 

Cheers
